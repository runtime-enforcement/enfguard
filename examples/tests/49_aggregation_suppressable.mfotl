ALWAYS (FORALL s, y. A(s) IMPLIES NOT (s <- SUM(x; y; B(x) AND C(y))))