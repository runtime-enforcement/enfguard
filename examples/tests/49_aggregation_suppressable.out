The formula
 □[0s,∞) (∀s:int. ∀y:int. A(s:int) → ¬(s:int <- SUM(x:int; y:int; C(y:int) ∧ B(x:int))))
is enforceable and types to
 □[0s,∞) ((∀s:int. (∀y:int. (A(s:int) →:R (¬((((∃x:int. ((C(y:int) : Sup) ∧:L B(x:int) : Sup)) : Sup) ∧:L (s:int <- SUM(x:int; y:int; B(x:int) ∧ C(y:int)))) : Sup) : Cau) : Cau) : Cau)) : Cau) : Cau
[Enforcer] @1 OK.
[Enforcer] @1 nothing to do proactively.
[Enforcer] @2 reactively commands:
Suppress:
C(2)
OK.
[Enforcer] @2 nothing to do proactively.
[Enforcer] @3 OK.
